### PR TITLE
Fix Weighted SSL

### LINKED ISSUE #2759: Weighted SSL model not unfreezable
### Describe the bug

In our HF Weighted SSL model implementation, we `detach()` the hidden states, meaning weights are not updated.

Relevant code:
https://github.com/speechbrain/speechbrain/blob/develop/speechbrain/lobes/models/huggingface_transformers/weighted_ssl.py#L81

```
hidden_states = torch.stack(feats.hidden_states, dim=0).detach()
```

### Expected behaviour

If passing `freeze=False` I'd expect the weights to get updated.

### To Reproduce

_No response_

### Environment Details

_No response_

### Relevant Log Output

_No response_

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by pplantinga**
Closes #2759 

I noticed a few issues with our Weighted SSL implementation, this should address them.

### CODE REVIEW

File: speechbrain/lobes/models/huggingface_transformers/weighted_ssl.py [hunk 1]
```diff
@@ -78,21 +85,28 @@ def forward(self, wav, wav_lens=None):
...
-        norm_weights = torch.nn.functional.softmax(self.weights, dim=-1)
+
         # Layernorming the layers representations if asked
         if self.layernorm:
-            hidden_states = [
-                F.layer_norm(t, (t.shape[-1],)) for t in hidden_states
-            ]
+            hidden_states = torch.stack(
+                [F.layer_norm(t, (t.shape[-1],)) for t in hidden_states], dim=0
+            )
```
[the remark this answers is not in the document]
Comment (pplantinga) [reply speechbrain/lobes/models/huggingface_transformers/weighted_ssl.py:102]: Lol that makes much more sense

**Review by Adel-Moumen — APPROVED** [review]
LGTM! Thanks @pplantinga very good work (and good spot :))
